optuna/trial.py

```diff
@@ -110,9 +110,10 @@ class Trial(BaseTrial):
     This object is passed to an objective function and provides interfaces to get parameter
     suggestion, manage the trial's state, and set/get user-defined attributes of the trial.
 
-    Note that this object is seamlessly instantiated and passed to the objective function behind
-    :func:`optuna.study.Study.optimize()` method (as well as optimize function); hence, in typical
-    use cases, library users do not care about instantiation of this object.
+    Note that the direct use of this constructor is not recommended.
+    This object is seamlessly instantiated and passed to the objective function behind
+    :func:`optuna.study.Study.optimize()` method (as well as optimize function); hence
```

Comment:
Some minors, not really introduced by this change.
```suggestion
    the :func:`optuna.study.Study.optimize()` method (as well as the optimize function); hence
```

Reply:
Thanks!